src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/BDD.cs

```diff
@@ -160,20 +135,18 @@ public ulong GetMin()
                 }
             }
 
-            return res;
+            return result;
         }
 
-        /// <summary>
-        /// O(1) operation that returns the precomputed hashcode.
-        /// </summary>
+        /// <summary>Gets a hashcode for the BDD.</summary>
         public override int GetHashCode() => _hashcode;
 
-        /// <summary>
-        /// A shallow equality check that holds if ordinals are identical and one's are identical and zero's are identical.
-        /// This equality is used in the _bddCache lookup.
-        /// </summary>
-        public override bool Equals(object? obj) =>
-            obj is BDD bdd &&
+        /// <summary>A shallow equality check that holds if ordinals are identical and one's are identical and zero's are identical.</summary>
+        public override bool Equals(object? obj) => Equals(obj as BDD);
+
+        /// <summary>A shallow equality check that holds if ordinals are identical and one's are identical and zero's are identical.</summary>
+        public bool Equals(BDD? bdd) =>
+            bdd is not null &&
```

Comment:
NIT: I believe we don't have a test where we pass null, might it be worth adding a quick unittest for that?